fastai/vision/learner.py

```diff
@@ -114,6 +114,18 @@ def _cl_int_plot_top_losses(self, k, largest=True, figsize=(12,12)):
         cl = int(cl)
         im.show(ax=axes.flat[i], title=
             f'{classes[self.pred_class[idx]]}/{classes[cl]} / {self.losses[idx]:.2f} / {self.probs[idx][cl]:.2f}')
+        if(heatmap):
+            xb,_ = self.data.one_item(im)
+            xb = xb.cuda()
```

Comment:
This is dangerous as you may break environments where people are on cpu only.  On the line above, you need to pass `detach=False, denorm=False` and xb won't be put on the CPU (on won't be denormalized, which is also needed), then this line won't be necessary anymore.

Reply:
This is also fixe now.